connexion/security/security_handler_factory.py

```diff
@@ -335,9 +333,7 @@ def wrapper(request):
 
     def _need_to_add_context_or_scopes(self, func):
         arguments, has_kwargs = inspect_function_arguments(func)
-        need_context = self.pass_context_arg_name and (
-            has_kwargs or self.pass_context_arg_name in arguments
-        )
+        need_context = has_kwargs or self.context_kw in arguments
```

Comment:
We could do this by removing the `has_kwargs` here.

Reply:
Ok i'll fix the two things as suggested